pkg/util/util.go

```diff
@@ -0,0 +1,114 @@
+// Package util provides utility functions for the sops-compliance-checker.
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+// Rule represents a single rule in the configuration.
+type Rule struct {
+	AllOf       []Rule `json:"allOf,omitempty"`
+	AnyOf       []Rule `json:"anyOf,omitempty"`
+	Match       string `json:"match,omitempty"`
+	Not         *Rule  `json:"not,omitempty"`
+	OneOf       []Rule `json:"oneOf,omitempty"`
+	Description string `json:"description,omitempty"`
+	URL         string `json:"url,omitempty"`
+}
+
+// Config represents the configuration for the sops-compliance-checker.
+type Config struct {
+	Rules []Rule `json:"rules"`
+}
+
+// LoadConfig loads the configuration from a JSON file.
+func LoadConfig(filePath string) (*Config, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	bytes, err := io.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+
+	var config Config
+	if err := json.Unmarshal(bytes, &config); err != nil {
+		return nil, err
+	}
+
+	if err := ValidateConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &config, nil
+}
+
+func bool2int(b bool) int {
+	if b {
+		return 1
+	}
+	return 0
+}
+
+// ValidateRule validates a rule.
+func ValidateRule(rule *Rule) error {
+
+	matchConditions := (bool2int(rule.Match != "") +
+		bool2int(rule.Not != nil) +
+		bool2int(len(rule.AllOf) > 0) +
+		bool2int(len(rule.AnyOf) > 0) +
+		bool2int(len(rule.OneOf) > 0))
+
+	if matchConditions != 1 {
+		return fmt.Errorf("Rule must exactly one match condition, got %d", matchConditions)
+	}
+
+	nestedRules := [][]Rule{
+		rule.AllOf,
+		rule.AnyOf,
+		rule.OneOf,
+	}
+
+	if rule.Not != nil {
+		if err := ValidateRule(rule.Not); err != nil {
+			return err
+		}
+	}
+
+	for _, rules := range nestedRules {
+		for _, subRule := range rules {
+			if err := ValidateRule(&subRule); err != nil {
+				return err
+			}
+		}
+	}
+
+	if rule.Match == "" &&
+		rule.Not == nil && len(rule.AllOf) == 0 && len(rule.AnyOf) == 0 && len(rule.OneOf) == 0 {
+		return errors.New("Rule must have a match condition or nested rules")
+	}
+
+	return nil
+}
+
+// ValidateConfig validates a configuration.
+func ValidateConfig(config *Config) error {
+	// Check that there is exactly one top-level rule
+	if len(config.Rules) != 1 {
```

Comment:
As discussed via chat, I would lift this restriction and treat the top-level rules like and `allOf` rule. That means, it is allowed to have 0 or more rules and we should validate all of them.

Reply:
Okay, I thought it makes sense to have at least one rule. I removed that restriction but I'm not sure if it makes sense to have config with 0 rules though